frontend/survey-project/src/app/components/pages/survey/pages/answer-survey-page/ui/answer-survey-email/answer-survey-email.component.ts

```diff
@@ -0,0 +1,136 @@
+import { Component } from '@angular/core';
+import { AnswerSurveyEmailRequest } from '../../../../../../../shared/dto/requests/survey/answer-survey-email-request';
+import { SurveyDetailsRequest } from '../../../../../../../shared/dto/requests/survey/survey-details-request';
+import { SurveyDetailsResponse } from '../../../../../../../shared/dto/responses/survey/survey-details-response';
+import { ActivatedRoute, Router } from '@angular/router';
+import { SurveyService } from '../../../../../../../shared/services/survey.service';
+import { CommonDialogsService } from '../../../../../../../shared/services/commondialog.service';
+import { ToastService } from '../../../../../../../shared/services/toast.service';
+import { LoaderService } from '../../../../../../../shared/services/loader.service';
+import { AnswerSurveyEmailResponse } from '../../../../../../../shared/dto/responses/survey/answer-survey-email-response';
+
+@Component({
+  selector: 'app-answer-survey-email',
+  templateUrl: './answer-survey-email.component.html',
+  styleUrl: './answer-survey-email.component.scss'
+})
+export class AnswerSurveyEmailComponent {
+  surveyTitle!: string;
+  surveyQuestion!: string;
+  emailId!: string;
+  surveyId!: string;
+  responseId!: string;
+  selectedAnswer!: string | null;
+  isAnswerConfirmed: boolean = false;
+  isDialogOpen: boolean = false;
+  isAnonymous!: boolean;
+  endDate!: string;
+  endDateTime: Date | null = null;
+  isClosed!: boolean;
+  isUserClosed!: boolean;
+
+  constructor(
+    private readonly route: ActivatedRoute,
+    private readonly surveyService: SurveyService,
+    private readonly dialogService: CommonDialogsService,
+    private readonly toastService: ToastService,
+    private readonly loaderService: LoaderService
+  ) {}
+
+  ngOnInit(): void {
+    this.emailId = this.route.snapshot.paramMap.get('email_id')!;
+    this.emailId = this.emailId?.replace(/[<>]/g, '').split(' ')[1] ?? this.emailId;
+
+    this.surveyId = this.route.snapshot.paramMap.get('survey_id')!;
+    this.responseId = this.route.snapshot.paramMap.get('response_id')!;
+    const initialAnswer = this.route.snapshot.paramMap.get('option');
+
+    if (initialAnswer){
+      this.selectAnswer(initialAnswer);
+    }
+
+    this.fetchSurveyDetails();
+  }
+
+  private fetchSurveyDetails(): void {
+    const request: SurveyDetailsRequest = {
+      survey_id: this.surveyId
+    };
+    this.loaderService.startLoading();
+    this.surveyService.getSurveyDetails(request).subscribe({
+      next: (response: SurveyDetailsResponse) => {
```

Comment:
You don't have to fix this, but we tend to use object that encapsulate all of the fields for an entity, so that we don't have a lot of fields that are tidally coupled with this component, but it's just entity this component uses.

Reply:
This is maybe not on the right place, but I'm talking about all of the variables that are not class tied but could be separate entity.